src/toil/job.py

```diff
@@ -92,6 +91,27 @@ def __init__(self, memory=None, cores=None, disk=None, preemptable=None, checkpo
         self._rvs = collections.defaultdict(list)
         self._promiseJobStore = None
 
+    @staticmethod
+    def _parseResource(value, resType):
+        """
+        Parse a value passed as resource for a Toil job.
+
+        :param value: The value passed as the resource
+        :param resType: The resource type
+        :return: parsed value
+        :rtype: int|float
+        """
+        assert resType in ('memory', 'disk', 'cores')
+        if value is None or isinstance(value, int):
+            return value
+        elif resType in ('memory', 'disk') and not isinstance(value, str):
+            raise RuntimeError('%s can only be passed int or humanized strings (e.g. 10G).'
+                               'Cannot handle type <%s>' % (resType, type(value).__name__))
+        elif resType == 'cores' and not isinstance(value, float):
+            raise RuntimeError('cores can only be passed int or float. Cannot handle type <%s>'
+                               % type(value).__name__)
+        else:
+            return human2bytes(str(value))
```

Comment:
Only strings should go to h2b. So I would assert that and remove the `str()` invocation.